src/main/resources/xslt/mods2solr.xsl

```diff
@@ -1,666 +1,555 @@
+            <f:array key="subject_date_of_birth">
+              <xsl:for-each select="m:subject/m:name">
+                <f:string>
+                  <xsl:value-of select="substring-before(m:namePart[@type='date'], '/')"/>
+                </f:string>
+              </xsl:for-each>
+            </f:array>
+            <f:array key="subject_date_of_death">
+              <xsl:for-each select="m:subject/m:name">
+                <f:string>
+                  <xsl:value-of select="substring-after(m:namePart[@type='date'], '/')"/>
+                </f:string>
+              </xsl:for-each>
+            </f:array>
+          </xsl:if>
+          <xsl:if test="m:subject/m:name/m:namePart[@type='termsOfAddress'] and m:subject/m:name/m:namePart[@type='termsOfAddress'] != ''">
+            <f:array key="subject_terms_of_address">
+              <xsl:for-each select="m:subject/m:name">
+                <f:string>
+                  <xsl:value-of select="f:replace(m:namePart[@type='termsOfAddress'], 'zh\|', '')"/>
+                </f:string>
+              </xsl:for-each>
+            </f:array>
+          </xsl:if>
+        </xsl:if>
+        <!-- Display label can in theory contain anything.
+             Resource Description seems to be the field that contains the most precise description of a given resource. -->
+        <xsl:if test="m:typeOfResource[@displayLabel='Resource Description']">
+          <f:string key="resource_description">
+            <xsl:value-of select="m:typeOfResource[@displayLabel='Resource Description']"/>
+          </f:string>
+        </xsl:if>
+        <!-- Type of resource gets extracted here if present -->
+        <!-- TODO: Should probably be named something like resource_categories -->
+        <xsl:if test="m:typeOfResource">
+          <f:array key="type_of_resource">
+            <xsl:for-each select="m:typeOfResource">
+              <f:string>
+                <xsl:value-of select="."/>
+              </f:string>
+            </xsl:for-each>
+          </f:array>
+        </xsl:if>
+        <!-- Create target_audience. This is not in our test files but part of MODS standard.
+              Could be used in other collections. -->
+        <xsl:if test="m:targetInfo">
+          <f:string key="target_audience">
+            <xsl:value-of select="m:targetInfo"/>
+          </f:string>
+        </xsl:if>
 
-  
-    <xsl:if test="string-length($img) &gt; 0">
-<!--      <f:map>
-        <xsl:if test="string-length($chapter) &gt; 0">
-          <f:string key="title">
-            <xsl:value-of select="$chapter"/>
+      </xsl:for-each>
+      <!--- This is the METS element with image metadata from the PremisObject
+       This extraction is used to provide information on image size-->
+      <xsl:for-each select="//mets:amdSec/mets:techMD[@ID='PremisObject1']//premis:objectCharacteristics">
+        <xsl:if test="premis:size">
+          <f:string key="file_size">
```

Comment:
`file_size` is defined as an int in `schema.xml`. It should be a long.

Reply:
Again correct. 
Victor?

Reply:
Updated